File: frontend/sakai-react/src/service/cadastros/CidadeService.js
import Axios from "axios";

export class CidadeService {
    url = process.env.REACT_APP_URI_API;

    cidades(){
        return Axios.get(this.url+"/cidade/");
    }

    inserir(objeto){
        return Axios.get(this.url+"/cidade/", objeto);
    }

    alterar(objeto){
        return Axios.get(this.url+"/cidade/", objeto);
    }

    excluir(id){
        return Axios.get(this.url+"/cidade/"+id);
    }
}

File: backend/src/main/java/com/dev/backend/service/cidade/CidadeService.java
package com.dev.backend.service.cidade;

import java.util.List;

import com.dev.backend.entity.Cidade;
import com.dev.backend.exception.InfoException;

public interface CidadeService {
    List<Cidade> buscarTodos();

    Cidade inserir(Cidade cidade) throws InfoException;

    Cidade alterar(Long id, Cidade cidade) throws InfoException;

    void excluir(Long id) throws InfoException;
}
